### ISSUE TITLE
Loss of color when using bat command to display highlighted source code in code block

### ISSUE THREAD

**Issue by stark71** [jaspervdj/patat#131]
Since it is not possible (AFAIK) to directly import a file for displaying it highlighted in a slide, I am looking for alternative solutions.

I have created a bash evaluating code block that uses the "bat" command to display the contents of a source file with syntax highlighting. However, the colors are lost in the output.

This can be a problem when wanting to display the code in a slide without having to manually copy and paste the code. Is there a solution to this problem that will allow the code to be displayed with colors intact?

```
```bash
bat file.c
`` `
```

**Comment by jaspervdj** [comment]
I think what's happening here is that `bat` detects is writing to a file and not a TTY and therefore it chooses to turn off colours.  You can maybe workaround this with using something like `bat --color always`.

An alternative would be to add two classes using the `~~~~~{.class1 .class2}` syntax in pandoc markdown: one to run it as bash; another to highlight it as C.  This is a full example that works for me (though I agree it's a bit hacky):

```
---
patat:
  eval:
    run_bash:
      command: bash
      fragment: false
      replace: true
...

# Example

This is not a quote

~~~~~{.run_bash .c}
cat test.c
~~~~~
```

**Comment by stark71** [comment]
Again you are right again. If I use --color always the block gets replaced with the bat output, the only downside is the spacing between lines.

I ended using your alternative solution as it works like a charm, I only had to change .c for .cpp

It would be nice to have a new kind of block or directive to import (and highlight) source code file (or a section of it) and maybe a few options to high light some lines. ie


```import(source_file.c, show_line_numbers=true, highlighted_lines=2,10,15)` ``

Thank you!
